scripts/kfctl.sh

```diff
@@ -157,9 +107,63 @@ createNamespace() {
   fi
 }
 
+# Generate all required components
+customizeKsApp() {
+  ks param set ambassador platform ${KUBEFLOW_PLATFORM}
+  ks param set jupyter platform ${KUBEFLOW_PLATFORM}
+}
+
+ksApply() {
+  pushd ${KUBEFLOW_KS_DIR}
+
+  if [ "${PLATFORM}" == "minikube" ]; then
+    createNamespace
+  fi
+
+  set +e
+  O=$(ks env describe default 2>&1)
+  RESULT=$?
+  set -e
+
+  if [ "${RESULT}" -eq 0 ]; then
+    echo "environment default already exists"
+  else
+    ks env add default --namespace "${K8S_NAMESPACE}"
+  fi
+
+  # Create all the core components
+  ks apply default -c ambassador
+  ks apply default -c jupyter
+  ks apply default -c centraldashboard
+  ks apply default -c tf-job-operator
+  ks apply default -c metacontroller
+  ks apply default -c spartakus
+
+  # Reduce resource demands locally
+  if [ "${PLATFORM}" != "minikube" ]; then
+    ks apply default -c argo
+    ks apply default -c katib
+  fi
+
+  popd
+
+  set +x
+  if [ "${PLATFORM}" == "minikube" ]; then
+    if is_kubeflow_ready; then
+      mount_local_fs
+      setup_tunnels
+    else
+      echo -e "${RED}Unable to get kubeflow ready${NC}"
+    fi
+  fi
+  set -x
+}
+
+
 if [ "${COMMAND}" == "init" ]; then
   DEPLOYMENT_NAME=${WHAT}
 
+  # Parse all command line options
```

Comment:
Should we move the code for argument parsing into a function 
parseArgs

and the remaining code into a function
named  main

and then just invoke main in the script?

Reply:
I think that'll make the code even cleaner.
I'll put those inside a function.